src/main/java/org/wildfly/security/sasl/digest/DigestClientFactory.java

```diff
@@ -60,6 +62,9 @@ public DigestClientFactory(final Provider provider) {
     public SaslClient createSaslClient(String[] mechanisms, String authorizationId, String protocol, String serverName,
             Map<String, ?> props, CallbackHandler cbh) throws SaslException {
 
+        if (cbh == null) throw ElytronMessages.log.mechCallbackHandlerMissing().toSaslException();
```

Comment:
I am not sure the compatibility is as important at this point - in both cases the end result is if you don't provide the required parameters the mechanism is unusable.